### PR TITLE
connection_template_resource

### PR DESCRIPTION
**Pull request by SHANDCRUZ**
### Description
Added Supported to Connection Template resource.

### Issues Resolved
NA

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
+# [v6.2.0]
```
Comment (AsisBagga) [inline CHANGELOG.md:4]: need to add unreleased

File: ov/connection_templates.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+	if err := json.Unmarshal(data, &connectionlist); err != nil {
+		return connectionlist, err
+	}
+	return connectionlist, nil
+}
+
+func (c *OVClient) UpdateConnectionTemplate(conntemplate ConnectionTemplate) error {
+	log.Infof("Initializing update of Connection Template for %s.", conntemplate.Name)
+	var (
+		uri = conntemplate.URI.String()
```
Comment (AsisBagga) [inline ov/connection_templates.go:99]: here are are we expecting user to provide the URI?

File: ov/connection_templates.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	}
+
+	log.Debugf("GetConnectionTemplate %s", data)
+	if err := json.Unmarshal(data, &connectionlist); err != nil {
+		return connectionlist, err
+	}
+	return connectionlist, nil
+}
+
+func (c *OVClient) UpdateConnectionTemplate(conntemplate ConnectionTemplate) error {
```
Comment (AsisBagga) [inline ov/connection_templates.go:96]: This function should provide a response body of type ConnectionTemplate and error

File: ov/connection_templates.go (hunk 2 above)
Comment (SHANDCRUZ) [reply ov/connection_templates.go:99]: its an update call right....so i just passed the object of the resource and there im getting the URI.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,13 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 
+# [v6.2.0 unreleased]
```
Comment (AsisBagga) [inline CHANGELOG.md:4]: please mention it like this
[v6.2.0] (unreleased)

**Review by AsisBagga — APPROVED** [review]
Apart from changelog, it LGTM.
Please update the changelog before merging.

File: ov/connection_templates.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	data, err := c.RestAPICall(rest.GET, uri, nil)
+	if err != nil {
+		return default_connection, err
+	}
+
+	log.Debugf("GetDefaultConnectionTemplate %s", data)
+	if err := json.Unmarshal(data, &default_connection); err != nil {
+		return default_connection, err
+	}
+	return default_connection, nil
```
Comment (nabhajit-ray) [inline ov/connection_templates.go:151]: Just a suggestion. Use variable names with mixed case. https://golang.org/doc/effective_go#mixed-caps

**Review by nabhajit-ray — APPROVED** [review]

**Pull request merged**